astlib/migrate_500_501.ml

```diff
@@ -297,7 +287,12 @@ and copy_value_binding :
              { ptyp_desc = Ptyp_poly (args_tyvars, rt) } );
        ppat_attributes = [];
       } ->
-          Some (pat, args_tyvars, rt)
+          Some (pat, args_tyvars, rt, `Var)
+      | {
+       Ast_500.Parsetree.ppat_desc = Ppat_constraint (pat, rt);
+       ppat_attributes = [];
+      } ->
+          Some (pat, [], rt, `NonVar)
```

Comment:
This change is too lax: we really want this branch to only be taken when `pat.ppat_desc` is not `Ppat_var`. Otherwise, we group the case
```ocaml
let (x,y) : int * int = ...
```
(which may have come from a value constraint or not) with 
```ocaml
let (x:int) = ...
```
which would have been desugared to `let (x: . int) = ... `  (with a `Ptyp_poly([], _)`) if it was a value constraint in 5.0.0 .